2-mandatory/2-financial-times.js

```diff
@@ -5,37 +5,85 @@
     Implement the function below, which will return a new array containing only article titles which will fit.
 */
 function potentialHeadlines(allArticleTitles) {
-    // TODO
+    let titleCharacters = [];
+    
+    for (let title of allArticleTitles) {
+    
+            if (title.length<=65) {
+             titleCharacters.push(title);
+        
+        }
+
+    }
+    return titleCharacters;
 }
 
+    
+
+
 /*
     The editor of the FT likes short headlines with only a few words!
     Implement the function below, which returns the title with the fewest words.
     (you can assume words will always be seperated by a space)
 */
 function titleWithFewestWords(allArticleTitles) {
-    // TODO
+    
+    let shortHeadlines;
+    let fewestWords;
+
+    for (let title of allArticleTitles) {
+        let numberOfWords = title.split(" ").length;
+   
+    if (fewestWords === undefined || numberOfWords < fewestWords) {
+        fewestWords = numberOfWords;
+        shortHeadlines = title;
+    }
+}
+    return shortHeadlines;
 }
 
+    
+
+
 /*
     The editor of the FT has realised that headlines which have numbers in them get more clicks!
     Implement the function below to return a new array containing all the headlines which contain a number.
     (Hint: remember that you can also loop through the characters of a string if you need to)
 */
-function headlinesWithNumbers(allArticleTitles) {
-    // TODO
+function findingNumber (str){
+        return /[0-9]/.test(str);
+}
+function headlinesWithNumbers(allArticleTitles){
+    let titleWithNumbers = [];
+    for (let headline of allArticleTitles) {
+        if (findingNumber(headline)=== true){
```

Comment:
When you have an if statement that looks like this: `if (findingNumber(headline)=== true){`, you can usually re-write it to be `if (findingNumber(headline)){`.
Can you think of why that works?